src/messaging/ExchangeContext.cpp

```diff
@@ -79,42 +79,10 @@ void ExchangeContext::SetResponseTimeout(Timeout timeout)
 CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgType, PacketBufferHandle msgBuf,
                                         const SendFlags & sendFlags)
 {
-    CHIP_ERROR err                         = CHIP_NO_ERROR;
-    Transport::PeerConnectionState * state = nullptr;
+    CHIP_ERROR err = CHIP_NO_ERROR;
 
     VerifyOrReturnError(mExchangeMgr != nullptr, CHIP_ERROR_INTERNAL);
 
-    state = mExchangeMgr->GetSessionMgr()->GetPeerConnectionState(mSecureSession);
-
-    // If a group message is to be transmitted to a destination node whose message counter is unknown.
-    if (state != nullptr && ChipKeyId::IsAppGroupKey(state->GetLocalKeyID()) && !state->IsPeerMsgCounterSynced())
```

Comment:
Where we trigger the MCSP for outgoing messages? It looks this scenario  is removed from this PR, if so, I'd expect to see obvious BSS decrease. is this logic moved to another place?

Reply:
Yes, I have discussed with Martin and Jerry. We are going to change the spec to remove this scienario.

https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/2491

Reply:
Let me rebase this over #6333, and remove packetHeader in the retrans table, then check if .bss can decrease.